src/main/proto/com/kodypay/grpc/ecom/v1/ecom.proto

```diff
@@ -62,28 +62,61 @@ message PaymentDetailsRequest {
 
 message PaymentDetailsResponse {
   oneof result {
-    Response response = 1;
+    PaymentDetails response = 1;
     Error error = 2;
   }
 
-  message Response {
-    string payment_id = 1; // The unique identifier created by Kody
-    string payment_reference = 2; // Your unique payment reference that was set during the initiation
-    string order_id = 3; // Your identifier of the order. It doesn't have to be unique, for example when the same order has multiple payments.
-    optional string order_metadata = 4; // A data set that can be used to store information about the order and used in the payment details. For example a JSON with checkout items. It will be useful as evidence to challenge chargebacks or any risk data.
-    PaymentStatus status = 5;
-    optional string payment_data_json = 6; // json blob containing payment data
-
-    google.protobuf.Timestamp date_created = 7;
-    optional google.protobuf.Timestamp date_paid = 8;
-
-    enum PaymentStatus {
-      PENDING = 0;
-      SUCCESS = 1;
-      FAILED = 2;
-      CANCELLED = 3;
+  message Error {
+    Type type = 1;
+    string message = 2;
+
+    enum Type {
+      UNKNOWN = 0;
+      NOT_FOUND = 1;
+      INVALID_REQUEST = 2;
     }
   }
+}
+
+message PaymentDetails {
+  string payment_id = 1; // The unique identifier created by Kody
+  string payment_reference = 2; // Your unique payment reference that was set during the initiation
+  string order_id = 3; // Your identifier of the order. It doesn't have to be unique, for example when the same order has multiple payments.
+  optional string order_metadata = 4; // A data set that can be used to store information about the order and used in the payment details. For example a JSON with checkout items. It will be useful as evidence to challenge chargebacks or any risk data.
+  PaymentStatus status = 5;
+  optional string payment_data_json = 6; // json blob containing payment data
+
+  google.protobuf.Timestamp date_created = 7;
+  optional google.protobuf.Timestamp date_paid = 8;
+
+  enum PaymentStatus {
+    PENDING = 0;
+    SUCCESS = 1;
+    FAILED = 2;
+    CANCELLED = 3;
+  }
+}
+
+message PaymentsRequest {
+  string store_id = 1;
+  // can be used to make pagination stable
+  // after first request, pass up the created date of the first entry
+  // then if a new payment occurs during page traversal, it will be ignored
+  optional google.protobuf.Timestamp created_before = 2;
+  int64 page = 3;
+  int64 page_size = 4;
+  optional string order_id = 5;
+}
+message PaymentsResponse {
```

Comment:
But the package will be different? Maybe here I just chosen a bad name given in all other circumstances we have top level messages for rpc requests/response? If the problem is the name maybe I can use `GetPayments` instead of `Payments` as the rpc name and then use `GetPaymentsRequest/Response` for the relevant rpc messages